app/models/trip_searcher.rb

```diff
@@ -0,0 +1,56 @@
+class TripSearcher
+  include ActiveModel::SerializerSupport
+
+  DEFAULT_PER_PAGE = 20
+
+  attr_reader :per_page, :page
+
+  def initialize(params)
+    @params = params
+    @per_page = positive_or_default(params[:per_page].to_i, DEFAULT_PER_PAGE)
+    @page = positive_or_default(params[:page].to_i, 1)
+  end
+
+  def results
+    @results ||= paginated_results
+  end
+
+  def total_results
```

Comment:
We should use Kaminari for this stuff. It makes all of that a lot easier. I think @tmecklem integrated Kaminari on another branch already.